go/controller/internal/httpserver/handlers/modelconfig.go

```diff
@@ -275,31 +254,43 @@ func (h *ModelConfigHandler) HandleCreateModelConfig(w ErrorResponseWriter, r *h
 
 	if providerConfigErr != nil {
 		log.Error(providerConfigErr, "Failed to assign provider config")
-		// Clean up the created secret if config assignment fails
-		log.V(1).Info("Attempting to clean up secret due to config assignment failure")
-		if providerTypeEnum != v1alpha1.Ollama {
-			if cleanupErr := h.KubeClient.Delete(r.Context(), secret); cleanupErr != nil {
-				log.Error(cleanupErr, "Failed to cleanup secret after config assignment failure")
-			}
-		}
 		w.RespondWithError(errors.NewBadRequestError(providerConfigErr.Error(), providerConfigErr))
 		return
 	}
 
 	if err := h.KubeClient.Create(r.Context(), modelConfig); err != nil {
-		log.Error(err, "Failed to create ModelConfig resource")
-		// If we fail to create the ModelConfig, we should clean up the secret
-		log.V(1).Info("Attempting to clean up secret after ModelConfig creation failure")
-		if providerTypeEnum != v1alpha1.Ollama {
-			if cleanupErr := h.KubeClient.Delete(r.Context(), secret); cleanupErr != nil {
-				log.Error(cleanupErr, "Failed to cleanup secret after ModelConfig creation failure")
-			}
-		}
+		log.Error(err, "Failed to create model config")
 		w.RespondWithError(errors.NewInternalServerError("Failed to create model config", err))
 		return
 	}
 
-	log.Info("Successfully created model config", "name", req.Name)
+	blockOwnerDeletion := true
+	controller := true
+	ownerRef := &metav1.OwnerReference{
+		APIVersion:         v1alpha1.GroupVersion.String(),
+		Kind:               "ModelConfig",
+		Name:               modelConfig.Name,
+		UID:                modelConfig.UID,
+		BlockOwnerDeletion: &blockOwnerDeletion,
+		Controller:         &controller,
+	}
+
+	_, err = CreateSecret(r.Context(), h.KubeClient, modelConfigSpec.APIKeySecretRef, common.GetResourceNamespace(), map[string]string{modelConfigSpec.APIKeySecretKey: req.APIKey}, ownerRef)
+	if err != nil {
+		log.Error(err, "Failed to create/update model config API key secret")
+		if cleanupErr := h.KubeClient.Delete(r.Context(), modelConfig); cleanupErr != nil {
+			log.Error(cleanupErr, "Failed to cleanup ModelConfig after secret creation failure")
+		}
+		if k8serrors.IsAlreadyExists(err) {
+			w.RespondWithError(errors.NewConflictError("Model config API key secret already exists", err))
+			return
+		}
+		w.RespondWithError(errors.NewInternalServerError("Failed to create/update model config API key secret", err))
+		return
+	}
+	log.V(1).Info("Successfully created/updated model config API key secret")
+
+	log.Info("Model config created successfully")
```

Comment:
Why was this block moved out of the conditionals above? I believe they were there for a reason. @peterj may remember

Reply:
out of which conditionals?